src/workerd/api/actor-state.c++

```diff
@@ -582,17 +582,24 @@ jsg::Promise<jsg::Value> DurableObjectStorage::transaction(jsg::Lock& js,
     jsg::Function<jsg::Promise<jsg::Value>(jsg::Ref<DurableObjectTransaction>)> callback,
     jsg::Optional<TransactionOptions> options) {
   auto& context = IoContext::current();
-  auto txn = jsg::alloc<DurableObjectTransaction>(context.addObject(
-        cache->startTransaction()));
 
   struct TxnResult {
     jsg::Value value;
     bool isError;
   };
 
   return context.blockConcurrencyWhile(js,
```

Comment:
Will the current `jsg::AsyncContextFrame` be properly propagated with `context.blockConcurrencyWhile`?

Reply:
I'm not sure. However, if it's broken, it's broken for both old and new DO storage implementations. This seems orthogonal to this PR so I'd rather address it separately.